standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/columnstats/merge/DecimalColumnStatsMergerTest.java

```diff
@@ -180,47 +188,59 @@ public void testMergeNonNullValues() {
 
   @Test
   public void testDecimalCompareEqual() {
-    assertTrue(DECIMAL_3.equals(DECIMAL_3));
+    assertTrue(DECIMAL_3.equals(getDecimal("3", 3, 0)));
+    // the equals method does not check for numerical equality,
+    // e.g., DECIMAL_3 is not equal to getDecimal("3", 30, 1)
   }
 
   @Test
   public void testDecimalCompareDoesntEqual() {
     assertFalse(DECIMAL_3.equals(DECIMAL_5));
+    assertFalse(DECIMAL_3.equals(getDecimal("30", 3, -1)));
   }
 
-  @Test
-  public void testCompareSimple() {
-    DecimalColumnStatsDataInspector data1 = new DecimalColumnStatsDataInspector(DATA_3);
-    DecimalColumnStatsDataInspector data2 = new DecimalColumnStatsDataInspector(DATA_5);
-    assertEquals(DECIMAL_5, merger.mergeHighValue(merger.getHighValue(data1), merger.getHighValue(data2)));
+  private void checkMergedValue(Decimal low, Decimal high) {
+    Objects.requireNonNull(low);
+    Objects.requireNonNull(high);
+    assertTrue(MetaStoreServerUtils.decimalToDouble(low) < MetaStoreServerUtils.decimalToDouble(high));
+    var data1 = new DecimalColumnStatsDataInspector();
+    data1.setLowValue(low);
+    data1.setHighValue(low);
+    var data2 = new DecimalColumnStatsDataInspector();
+    data2.setLowValue(high);
+    data2.setHighValue(high);
+
+    assertEquals(low, merger.mergeLowValue(data1.getLowValue(), data2.getLowValue()));
+    assertEquals(low, merger.mergeLowValue(data2.getLowValue(), data1.getLowValue()));
+    assertEquals(high, merger.mergeHighValue(data1.getHighValue(), data2.getHighValue()));
+    assertEquals(high, merger.mergeHighValue(data2.getHighValue(), data1.getHighValue()));
```

Comment:
nit. IMHO a unit test should test only a single functionality / method. So `mergeHighValue` sould be moved to separate test methods like it was in the original code.